### PR TITLE
[VL] Optimize failure output for WrappedVeloxMemoryPool

### PR DESCRIPTION
**Pull request by zhztheplayer**

### CODE REVIEW

**Comment by zhztheplayer** [comment]
Before
```
java.lang.UnsupportedOperationException: Not enough spark off-heap execution memory. Acquired: 33554432, granted: 18874368. Try tweaking config option spark.memory.offHeap.size to get larger space to run this application. 
	at io.glutenproject.memory.alloc.GlutenManagedReservationListener.reserveOrThrow(GlutenManagedReservationListener.java:53)
	at io.glutenproject.vectorized.ArrowOutIterator.nativeHasNext(Native Method)
	at io.glutenproject.vectorized.ArrowOutIterator.hasNextInternal(ArrowOutIterator.java:48)
	at io.glutenproject.vectorized.GeneralOutIterator.hasNext(GeneralOutIterator.java:37)
	at io.glutenproject.backendsapi.glutendata.GlutenIteratorApi$$anon$2.hasNext(GlutenIteratorApi.scala:242)
	at io.glutenproject.vectorized.CloseableColumnBatchIterator.hasNext(CloseableColumnBatchIterator.scala:41)
	at org.apache.spark.InterruptibleIterator.hasNext(InterruptibleIterator.scala:37)
	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
	at org.apache.spark.shuffle.GlutenColumnarShuffleWriter.internalWrite(GlutenColumnarShuffleWriter.scala:96)
	at org.apache.spark.shuffle.GlutenColumnarShuffleWriter.write(GlutenColumnarShuffleWriter.scala:196)
	at org.apache.spark.shuffle.ShuffleWriteProcessor.write(ShuffleWriteProcessor.scala:59)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:99)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:52)
	at org.apache.spark.scheduler.Task.run(Task.scala:131)
	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:506)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1491)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:509)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:750)
```
After

```
E0412 09:48:26.668958   809 Exceptions.h:68] Line: /opt/gluten/cpp/velox/memory/VeloxMemoryPool.cc:116, Function:allocateZeroFilled, Expression:  allocateZeroFilled failed with 1 entries and 26774293 bytes each from Memory Pool[op.0.0.0.TableScan LEAF MALLOC], cause: Error during calling Java code from native code: java.lang.UnsupportedOperationException: Not enough spark off-heap execution memory. Acquired: 33554432, granted: 18874368. Try tweaking config option spark.memory.offHeap.size to get larger space to run this application. 
	at io.glutenproject.memory.alloc.GlutenManagedReservationListener.reserveOrThrow(GlutenManagedReservationListener.java:53)
	at io.glutenproject.vectorized.ArrowOutIterator.nativeHasNext(Native Method)
	at io.glutenproject.vectorized.ArrowOutIterator.hasNextInternal(ArrowOutIterator.java:48)
	at io.glutenproject.vectorized.GeneralOutIterator.hasNext(GeneralOutIterator.java:37)
	at io.glutenproject.backendsapi.glutendata.GlutenIteratorApi$$anon$2.hasNext(GlutenIteratorApi.scala:242)
	at io.glutenproject.vectorized.CloseableColumnBatchIterator.hasNext(CloseableColumnBatchIterator.scala:41)
	at org.apache.spark.InterruptibleIterator.hasNext(InterruptibleIterator.scala:37)
	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
	at org.apache.spark.shuffle.GlutenColumnarShuffleWriter.internalWrite(GlutenColumnarShuffleWriter.scala:96)
	at org.apache.spark.shuffle.GlutenColumnarShuffleWriter.write(GlutenColumnarShuffleWriter.scala:196)
	at org.apache.spark.shuffle.ShuffleWriteProcessor.write(ShuffleWriteProcessor.scala:59)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:99)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:52)
	at org.apache.spark.scheduler.Task.run(Task.scala:131)
	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:506)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1491)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:509)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:750)
, Source: RUNTIME, ErrorCode: MEM_ALLOC_ERROR
```

File: cpp/velox/memory/VeloxMemoryPool.cc [hunk 1]
```diff
@@ -99,16 +104,24 @@ class WrappedVeloxMemoryPool final : public velox::memory::MemoryPool {
...
-    if (!gluten_alloc_->AllocateZeroFilled(alignedSize, 1, &buffer)) {
-      VELOX_FAIL(
-          "WrappedVeloxMemoryPool: Failed to allocate (zero filled) " + std::to_string(alignedSize) + " members, " +
-          std::to_string(1) + " bytes for each")
-    }
-    if (FOLLY_UNLIKELY(buffer == nullptr)) {
+    try {
+      bool succeed = gluten_alloc_->AllocateZeroFilled(alignedSize, 1, &buffer);
+      if (!succeed) {
+        VELOX_FAIL(
```
Comment (jinchengchenghh) [inline cpp/velox/memory/VeloxMemoryPool.cc:110]: Can you use this format?
```
int64_t toSkip = position;
VELOX_FAIL("Seeking past end of ByteStream: {}", position);
```

Comment (zhztheplayer) [reply cpp/velox/memory/VeloxMemoryPool.cc]: Sure, will add a commit

File: ep/build-velox/src/get_velox.sh [hunk 2]
```diff
@@ -58,6 +58,7 @@ function process_setup_ubuntu {
   sed -i '/^sudo --preserve-env apt update && sudo apt install -y/a\  uuid-dev \\' scripts/setup-ubuntu.sh
   sed -i 's/^  liblzo2-dev.*/  liblzo2-dev \\/g' scripts/setup-ubuntu.sh
   sed -i '/libre2-dev/d' scripts/setup-ubuntu.sh
+  sed -i '/libgmock-dev/d' scripts/setup-ubuntu.sh # resolved by ep/build-velox/build/velox_ep/CMake/resolve_dependency_modules/gtest.cmake
```
Comment (zhztheplayer) [inline ep/build-velox/src/get_velox.sh:61]: @zhejiangxiaomai 

`libgmock` installed by the script is 1.10. The one installed by `resolve_dependency_modules/gtest.cmake` is 1.13.

I am removing his one. Any comment?

Comment (zhejiangxiaomai) [reply ep/build-velox/src/get_velox.sh:61]: let use 1.13

Comment (zhztheplayer) [reply ep/build-velox/src/get_velox.sh:61]: OK. So I suppose this change is reasonable?

Comment (zhejiangxiaomai) [reply ep/build-velox/src/get_velox.sh:61]: Yes.

**Review by zhejiangxiaomai — APPROVED** [review]

File: ep/build-velox/src/get_velox.sh (hunk 2 above)
Comment (zhztheplayer) [reply ep/build-velox/src/get_velox.sh:61]: Got it. Thanks for helping.

**Pull request merged**